### DISCUSSION TITLE
Query Server

### DISCUSSION THREAD

**Discussion by gc-victor** [epicweb-dev/epic-stack#521] [Ideas]
Hi @kentcdodds!

I believe [Query](https://github.com/gc-victor/query) is a strong fit for the Epic Stack database part. It also provides[ Edge Functions](https://github.com/gc-victor/query?tab=readme-ov-file#function) without any extra configuration, and soon, I will add  assets feature too.

Once implemented, the user will benefit from the Query features and ecosystem, such as [Query CLI](https://github.com/gc-victor/query/blob/main/README.md#cli), [Query API](https://github.com/gc-victor/query/blob/main/README.md#apis) and  [Query Studio](https://github.com/gc-victor/query-studio).

Would you like to consider adding it? I can help you with the process.

**Comment by kentcdodds** [comment]
It looks very cool! I don't think that I'm ready to switch everything up just yet, but the best way to convince me this is the way to go is to make an example. Would you be interested in doing that? Check the docs for the best way to make an example.

Thanks for the idea!

**Reply by gc-victor** [reply]
Implementing it should be pretty straightforward as it is needed to add a couple of changes to the `Dockerfile` and change the executed command from the `litefs.yml` file to execute the `process.sh` file.

Dockerfile

```
# Download and installs Query Server
RUN curl --proto '=https' --tlsv1.2 -LsSf https://github.com/gc-victor/query/releases/latest/download/query-server-installer.sh | sh

# It will execute the Query Server and your App
COPY --link process.sh process.sh
RUN chmod +x process.sh

# Enable Query Server Proxy
ENV QUERY_SERVER_PROXY="true"
# Your App port
ENV QUERY_SERVER_PROXY_PORT="3001"
```

process.sh
```
#!/bin/bash

set -m
~/.cargo/bin/query-server &
npm start &
fg %1
```

litefs.yml

```
...
exec:
  - cmd: "./process.sh"
...
```

Then add a couple of secrets:

`fly secrets set QUERY_SERVER_TOKEN_SECRET=__USE_AN_ALPHANUMERIC_HASH__`

And 

`fly secrets set QUERY_SERVER_ADMIN_EMAIL=__USE_YOUR_EMAIL__`

And

`fly secrets set QUERY_SERVER_ADMIN_PASSWORD=__USE_A_SECURE_PASSWORD__`

**Reply by kentcdodds** [reply]
Awesome! I'll be excited to merge the PR for the [example](https://github.com/epicweb-dev/epic-stack/blob/main/docs/examples.md) :)

**Reply by gc-victor** [reply]
~I have it running, but, some actions doesn't work. It shows the message "No requestInfo found in root loader". Is it expected?~

Fixed! :tada: 

https://epic-appt-c191.fly.dev

**Reply by gc-victor** [reply]
This is the repository of the example https://github.com/gc-victor/epic-stack-with-query

**Reply by kentcdodds** [reply]
Merged. Thanks!
